lib/routes/account.js

```diff
@@ -973,78 +977,89 @@ module.exports = (log, db, mailer, Password, config, customs, signinUtils, push)
         }
         request.setMetricsFlowCompleteSignal(flowCompleteSignal)
 
-        return resetAccountData()
+        return checkRecoveryKey()
+          .then(resetAccountData)
+          .then(deleteRecoveryKey)
           .then(createSessionToken)
           .then(createKeyFetchToken)
           .then(recordSecurityEvent)
           .then(createResponse)
           .then(reply, reply)
 
+        function checkRecoveryKey() {
+          if (recoveryKeyId && wrapKb) {
+            return db.getRecoveryKey(accountResetToken.uid, recoveryKeyId)
+          }
+
+          return Promise.resolve()
```

Comment:
we have this weird pattern where we mix the bluebird `P.resolve` with native `Promise.resolve`. I'm not sure how I feel about that

Reply:
We should be using `P.resolve` 👍🏽